From 797026638f234a4a178c5ca6be8939df6ebd5afd Mon Sep 17 00:00:00 2001
From: Kyle Brenneman <kbrenneman@nvidia.com>
Date: Wed, 10 Dec 2025 08:31:56 -0700
Subject: [PATCH 17/18] Fix an incorrect return statement in eplWlSwapBuffers.

In eplWlSwapBuffers, if WaitForPreviousFrames fails, then do all the
necessary cleanup before returning.
---
 src/wayland/wayland-surface.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/wayland/wayland-surface.c b/src/wayland/wayland-surface.c
index f34489e..4905911 100644
--- a/src/wayland/wayland-surface.c
+++ b/src/wayland/wayland-surface.c
@@ -1348,7 +1348,7 @@ EGLBoolean eplWlSwapBuffers(EplPlatformData *plat, EplDisplay *pdpy,
     {
         if (!WaitForPreviousFrames(psurf))
         {
-            return EGL_FALSE;
+            goto done;
         }
     }
     else
-- 
2.51.0

